engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -906,6 +907,21 @@ - (void)viewWillAppear:(BOOL)animated {
   [super viewWillAppear:animated];
 }
 
+- (void)viewIsAppearing:(BOOL)animated {
+  TRACE_EVENT0("flutter", "viewIsAppearing");
+  if (self.engine.viewController == self) {
+    // The earliest the view can access the scene is in viewIsAppearing.
```

Comment:
It looks like the `UIWindowScene` first becomes accessible to the view in `[UIView didMoveToWindow]` (when establishing the view hierarchy), and `viewIsAppearing` is called by the layout process so it's a bit later than that:
```
  * frame #0: 0x000000018545a010 UIKitCore`-[UIViewController viewIsAppearing:]
    frame #1: 0x0000000185449ab8 UIKitCore`-[UIViewController __viewIsAppearing:skipWindowCheck:] + 188
    frame #2: 0x0000000186142cd4 UIKitCore`-[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1072
```
Not sure if that makes a difference tho.